pkg/integrations/v2/app_o11y_receiver/exporters/receiver_metrics.go

```diff
@@ -0,0 +1,60 @@
+package exporters
+
+import (
+	"context"
+
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/models"
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/utils"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ReceiverMetricsExporterConfig contains options for the ReceiverMetricsExporter
+type ReceiverMetricsExporterConfig struct {
+	Reg *prometheus.Registry
+}
+
+// ReceiverMetricsExporter is a app o11y receiver exporter that will capture metrics
+// about counts of logs, exceptions, measurements, traces being ingested
+type ReceiverMetricsExporter struct {
+	totalLogs         prometheus.Counter
+	totalMeasurements prometheus.Counter
+	totalExceptions   prometheus.Counter
+}
+
+// NewReceiverMetricsExporter creates a new ReceiverMetricsExporter
+func NewReceiverMetricsExporter(conf ReceiverMetricsExporterConfig) AppO11yReceiverExporter {
+	exp := &ReceiverMetricsExporter{
+		totalLogs: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: utils.MetricsNamespace,
+			Name:      "total_logs",
+			Help:      "Total number of ingested logs",
+		}),
+		totalMeasurements: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: utils.MetricsNamespace,
+			Name:      "total_measurements",
+			Help:      "Total number of ingested measurements",
+		}),
+		totalExceptions: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: utils.MetricsNamespace,
+			Name:      "total_exceptions",
+			Help:      "Total number of ingested exceptions",
+		}),
```

Comment:
nits:

1. Don't use Namespace from the opts, because that makes it harder to search through the codebase to find where a metric is defined. Instead, pass the full literal name in to the Name field.
2. The naming conventions here don't match Prometheus; the counters should be `logs_total`, not `total_logs`.

Reply:
- [ ] update